src/lib/button/button.spec.ts

```diff
@@ -210,3 +214,15 @@ class TestApp {
     this.clickCount++;
   }
 }
+
+class FakeViewportRuler {
```

Comment:
Can we move this to its own file instead of duplicating it? Having `fake-viewport-ruler.ts` under `overlay/position` would work.